src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,35 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type.equals(Bulldozer.class)) {
+            return new ArrayList<>(new BulldozerProducer().get());
+        } else if (type.equals(Excavator.class)) {
+            return new ArrayList<>(new ExcavatorProducer().get());
+        } else if (type.equals(Truck.class)) {
+            return new ArrayList<>(new TruckProducer().get());
+        }
+        return new ArrayList<>();
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, T value) {
+        machines.replaceAll(machine -> value);
+    }
+
+    @Override
+    public void startWorking(List<? extends Machine> machines) {
+        machines.forEach(Machine::doWork);
```

Comment:
You can use loop to increase readability, but I am not sure if this is important.

Reply:
This is the same as for each loop  but using java streams

Reply:
I understand, it is not a mistake, just proposition.